tasks/clay-approve.js

```diff
@@ -2,22 +2,32 @@
 task("clay-approve", "Approves clay token to given account")
     .addParam("spender", "The account's address")
     .addParam("amount", "The amount to be approved")
+    .addOptionalParam(
+        "gnosisSafe",
+        "Gnosis safe address, should be given if trasactions need to be submitted to gnosis",
+        undefined,
+        types.string
+    )
     .setAction(
         async (args, hre) => {
             const { expect } = require('chai');
             const { deployer } = await hre.getNamedAccounts();
             const { getTxUrl } = require('../utils/helper');
+            const submitTransactionToGnosisSafe = require('../gnosis/helper');
 
             const clayToken = await ethers.getContract("ClayToken", deployer);
             console.log("Using ClayToken: ", clayToken.address);
 
-            const clayBalance = await clayToken.balanceOf(deployer);
+            const clayBalance = await clayToken.balanceOf(args.gnosisSafe || deployer);
```

Comment:
TODO: need to improve this. Did you hear anything back from safe about using gnosis signer to call read only functions?

Reply:
nope